kubernetes/istio.go

```diff
@@ -248,6 +252,63 @@ func getStatus(statuses map[string][]byte) ([]*ProxyStatus, error) {
 	return fullStatus, nil
 }
 
+func (in *K8SClient) GetConfigDump(namespace, podName string) (*ConfigDump, error) {
+	// Fetching the config_dump data, raw.
+	resp, err := in.EnvoyForward(namespace, podName, "/config_dump")
+	if err != nil {
+		log.Errorf("Error fetching config_map: %v", err)
+		return nil, err
+	}
+
+	cd := &ConfigDump{}
+	err = json.Unmarshal(resp, cd)
+	if err != nil {
+		log.Errorf("Error Unmarshalling the config_dump: %v", err)
+	}
+
+	return cd, err
+}
+
+func (in *K8SClient) EnvoyForward(namespace, podName, path string) ([]byte, error) {
+	writer := new(bytes.Buffer)
+
+	clientConfig, err := ConfigClient()
+	if err != nil {
+		log.Errorf("Error getting Kubernetes Client config: %v", err)
+		return nil, err
+	}
+
+	// First try whether the pod exist or not
+	_, err = in.GetPod(namespace, podName)
+	if err != nil {
+		log.Errorf("Couldn't fetch the Pod: %v", err)
+		return nil, err
+	}
+
+	// Create a Port Forwarder
+	f, err := config_dump.NewPortForwarder(in.k8s.CoreV1().RESTClient(), clientConfig, namespace, podName, "localhost", "15000", writer)
```

Comment:
I'd try to put the host and portforward post in the config, for two reasons: one for document this, if you don't stop here, you may think this is exposed externally instead to run a port-forward locally, and second, considering that we expose as default on 20001 port, I'd put together all these ports in case that user change 20001 -> 15000 and things don't work as expected.

I know that this is unlikely be a problem, but just to prevent this in the future.

Reply:
Good point!

Reply:
Oh wait. This portforward is done in the kiali container, so it is not external and it doesn't depend on user's ports. It just depends on kiali's pod open ports. I'd have conflict with the `20001` port if the port-forwarding was performed in the client side.

I think that there is no need to add this in the config because it wouldn't affect to the user at any way.

Reply:
But in the same container Kiali is opening the 20001 port, isn't it ? It won't have a conflict if you change this to "20001" ?

Reply:
you are right. Had it wrong. I've pushed a commit adding the local port in the config.